### ISSUE TITLE
Mod only partially works, please help

### ISSUE THREAD

**Issue by Pangaea666** [QuietusPlus/EMC-Stock#1]
Love your mod, and would like to use it, but unfortunately I can't get it to work properly. I'm on a Linux system, but I'm not sure if that should matter, as the file structure looks similar/equal. 

I'm particularly interested in the 2nd level skills descriptions by default, but sadly it doesn't show up no matter what I do. I've tried stock and minimal, and also downloaded the file archive from Git, but that doesn't work either. The strange thing is that some changes take effect, but most don't. The intro movie is skipped and with the Minimal version the mysterious merchant has the Respec potion. The potion itself doesn't appear to work however, as I still had all skills points spent after drinking the potion.

Would also love the auto loot feature, but it doesn't work. Item descriptions don't change either, there isn't for instance (Crafting) in front of crafting items. The inventory screen is unchanged on my end. The reduced weight, however, takes effect when I used the Minimal version.

Could these problems be due to running the game on Linux (with the Linux build from GOG), or do you know if it's possible to get the mod working like intended with the most recent version of the game?

**Comment by QuietusPlus** [comment]
Not sure about Linux being a problem, my guess is that it should work the same...

Do you perhaps have any other mods installed?

**Comment by Pangaea666** [comment]
Originally I did have other mods installed (mainly improved graphics), however I disabled these (renamed folders, deleted files), and there was no change. Just to be sure, however, I have uninstalled and reinstalled the game, using only your mod, the minimal version. Sadly the same problem as before persists. A few things take effect (weight reduction, cutscene intro doesn't play, Respec potion added), but most don't. The Respec potion doesn't work (talents not 'freed', though I'm only slightly out of the training tree, at level 8). It looks like containers and other loot is supposed to light up, but it doesn't. The category name isn't added to the Inventory screen, and the second level talent descriptions are still missing. The other things don't worry me too much, but I would really like to have the extra info about talents, without having to resort to the wiki.

I find it odd that some things take effect while others do not. I would think either the whole thing worked, or nothing did. The structure actually does look slightly changed, but not sure if it matters. On the GOG Linux install there is a "game" folder, which contains "CookedPC". I didn't have a "base_scripts.dzip" file there either, so the game must just take it from the massive pack0.dzip instead.

Here is a picture of the file structure, in case it matters: https://i.imgur.com/9JujBHj.jpg

Would be excellent if you could help me sort out this, especially considering I see from Nexusmods that the mod isn't in active development any more. Understand if you are tied up with others matters though.

![tw2-linux](https://cloud.githubusercontent.com/assets/2002335/17629658/b8197c9e-60bb-11e6-9f03-b025b7ef5cc5.jpg)

I tried downloading the ui_character.swf file from Git and put it in /game/CookedPC/globals/gui, hoping it would be used instead of the original games file, but the extra talent descriptions don't show up then either.

**Comment by Pangaea666** [comment]
I've searched around some more, and unfortunately it looks like this mod simply won't work properly on Linux. Something CDPR did with RedKit means it doesn't transfer well on a Linux system, for whatever reason. 

Here are a few posts that hints about the problem:
https://www.gog.com/forum/the_witcher_2/install_full_combat_rebalance_mod_on_linux
http://forums.cdprojektred.com/threads/29941-The-Witcher-2-Assassins-of-Kings-Enhanced-Edition-available-on-Linux!/page17

From what I can see, XML changes take effect, but script changes do not. This is probably why features in your mod such as weight reduction and introduction of the Respec potion appear in my game, but script related stuff does not, such as the actual feature of the Respec potion, level 2 descriptions, categories in the inventory, or faster medallion use or longer-lasting corpses. I'd love some of this in my game, but sadly I don't think it's possible on a Linux system, due to how CDPR designed RedKit/the game.

**Comment by QuietusPlus** [comment]
I'm reading up on the stuff you've provided as we speak...

The reason it's not working is probably because the vanilla scripts don't get overwritten, so the question remains: Where are those vanilla scripts stored on the Linux build? 

Did you happen to come across anything specific? Or could you take a look inside those files, I unfortunately don't have access to the Linux version (I could extract the GOG installer myself if need be).

PS: This mod isn't really dependent on RedKit as far as I know. I've only used it for it's debugging abilities. The creation was entirely done using a text editor and third party tools to (un)pack base_scripts.dzip...

**Comment by Pangaea666** [comment]
The odd thing is that pack0 and DLCs are present in dzip form, but base_scripts.dzip isn't. Can't find it anywhere either. Just now I managed to extract the whole pack0 archive, hoping the scripts were in there, but no luck, so I'm actually not sure where the game takes its scripts from. I've installed RedKit via PlayOnLinux/wine, and there the script files are in /bin/scripts, but the /bin folder in the normal game installation location doesn't contain much, only a config folder.

Hopefully you are correct that we just need to overwrite the default script files, because then it should be possible to solve. The problem is that I can't see them anywhere.

I see some eON related files, which was used to 'port' the game to Linux, but that may not matter. 

edit: There is a file called compiledscripts.w2scripts. But this exists on the normal Windows installation too (in addition to base_scripts.dzip), and I get error messages when trying to unpack it with Gibbed RED Tools.

**Comment by QuietusPlus** [comment]
Would you mind doing a system wide search for .dzip? Maybe that'll provide some more information.

And here's just a quick thought:

Those eON files don't exist on Windows, they probably packaged most of what we are "missing"... These are related to RedKit packaging, correct?

I downloaded the Linux version but unfortunately have no way of installing/unpacking the .sh within windows.

Would really like to have a look at the differences myself, so I may end up doing this via a Virtual Machine.

EDIT: Got the VM set up, currently installing the game.

**Comment by Pangaea666** [comment]
eON is what was used to get the game over to Linux. Apparently it's a "wrapper" rather than a proper port. From what I read the game ran horribly for quite a while, but thankfully it runs fine presently on my current system, so I guess they eventually found out how to do that stuff. CDPR didn't do it themselves, but a company called Virtual Programming. No idea exactly what they did, but I don't think RedKit is out in Linux, either natively or using a wrapper. But I did get it installed and running via PlayOnLinux. I don't know how to use it, though.

This may look messy, but I printed all the files in the TW2EE folder. I've got a few mods installed now, so those files are included. It's the environment and weapon texture mods, weight increase on books, and weight reductions on low weight material (so I can pick up all the 'loot' my RPG-heart desires). I've also deleted a few files because I just can't stand barking dogs in games, so a few sound files are gone.

[TW2.txt](https://github.com/QuietusPlus/EMC-Stock/files/420649/TW2.txt)

edit: I did a system wide search btw, but couldn't find base_scripts.dzip anywhere in the linux install. Windows has that file, but for some reason the Linux version doesn't. I have many other .dzips, but not that one. If it's 'baked' into eON or some other file, I guess we're screwed :(

**Comment by QuietusPlus** [comment]
Finally done with the installation, took a hefty amount of time within a virtual machine. I'll soon start experimenting with different ideas I've had.

My best guess at the moment is that the Linux version only has a static `compiledscripts.w2scripts`, whereas the Windows version grabs the scripts from `base_scripts.dzip` and then compiles them each time the game is launched... Which would be a problem, because I'm not sure how to compile those myself.

If my theory turns out to be true we may be able to run the Windows version including this mod. We then take the compiled `compiledscripts.w2scripts` and copy it to Linux (this is a bit of a stretch)...

## File List - Clean Install

`...` = Removed similarly named files / filetypes

``` bash
├── docs
│   ├── End User License Agreement.txt
│   ├── installer_readme.txt
│   └── The Witcher 2 EE manual.pdf
├── game
│   ├── bin
│   │   └── config
│   │       ├── lang
│   │       │   ├── BR.ini
│   │       │   ├── CZ.ini
│   │       │   ├── DE.ini
│   │       │   └── ...
│   │       ├── profiles
│   │       │   ├── PCHigh.ini
│   │       │   ├── PCLow.ini
│   │       │   ├── PCMedium.ini
│   │       │   └── PCUber.ini
│   │       ├── Community.ini
│   │       ├── DIMapping.ini
│   │       ├── Input_AZERTY.ini
│   │       ├── Input_QWERTY.ini
│   │       ├── Input_QWERTZ.ini
│   │       ├── Rendering.ini
│   │       └── User.ini
│   ├── CookedPC
│   │   ├── globals
│   │   │   └── gui
│   │   │       ├── fonts
│   │   │       │   ├── bll55.otf
│   │   │       │   └── bll75.otf
│   │   │       ├── loadingscreens
│   │   │       │   ├── 00_default.swf
│   │   │       │   ├── 00_loading01.swf
│   │   │       │   ├── 00_loading.swf
│   │   │       │   └── ...
│   │   │       └── savescreens
│   │   │           ├── l01.bmp
│   │   │           ├── l02.bmp
│   │   │           ├── l03.bmp
│   │   │           └── ...
│   │   ├── movies
│   │   │   ├── 1_rivia_mem.usm
│   │   │   ├── 2_apple_island_mem.usm
│   │   │   ├── 308jaskier.usm
│   │   │   └── ...
│   │   ├── sounds
│   │   │   ├── l01_cs
│   │   │   │   ├── cs000_foltest.fsb
│   │   │   │   ├── cs000_foltest_get_in_siege_tower.fsb
│   │   │   │   ├── cs000_foltest_shilard.fsb
│   │   │   │   └── ...
│   │   │   ├── l02_cs
│   │   │   │   ├── cs101_iorweth_cliff.fsb
│   │   │   │   ├── cs105_tentacle.fsb
│   │   │   │   ├── cs106_ambush.fsb
│   │   │   │   └── ...
│   │   │   ├── l03_cs
│   │   │   │   ├── cs200_healing_saskia.fsb
│   │   │   │   ├── cs201_geralt_anathema.fsb
│   │   │   │   ├── cs201_owl_battle.fsb
│   │   │   │   └── ...
│   │   │   ├── l04_cs
│   │   │   │   ├── cs301_detmold_in_keadwen.fsb
│   │   │   │   ├── cs301_geralt_take_shilard_hostage.fsb
│   │   │   │   ├── cs302_capture.fsb
│   │   │   │   └── ...
│   │   │   ├── music
│   │   │   │   ├── l01_music.fev
│   │   │   │   ├── l01_music.fsb
│   │   │   │   ├── l02_music.fev
│   │   │   │   └── ...
│   │   │   ├── animals_cat.fsb
│   │   │   ├── animals_chicken.fsb
│   │   │   ├── animals_common.fsb
│   │   │   └── ...
│   │   ├── abetterui.dzip
│   │   ├── alchemy_suit.dzip
│   │   ├── arena.dzip
│   │   ├── compiledscripts.w2scripts
│   │   ├── cz0.w2strings
│   │   ├── darkdiff.dzip
│   │   ├── de0.w2speech
│   │   ├── de0.w2strings
│   │   ├── dlc_finishers.dzip
│   │   ├── elf_flotsam.dzip
│   │   ├── en0.w2speech
│   │   ├── en0.w2strings
│   │   ├── es0.w2strings
│   │   ├── fr0.w2speech
│   │   ├── fr0.w2strings
│   │   ├── hairdresser.dzip
│   │   ├── harpy_feathers.dzip
│   │   ├── hu0.w2strings
│   │   ├── it0.w2strings
│   │   ├── jp0.w2strings
│   │   ├── krbr.dzip
│   │   ├── magical_suit.dzip
│   │   ├── merchant.dzip
│   │   ├── pack0.dzip
│   │   ├── pack0.dzip.split00
│   │   ├── pack0.dzip.split01
│   │   ├── pack0.dzip.split02
│   │   ├── pl0.w2speech
│   │   ├── pl0.w2strings
│   │   ├── roche_jacket.dzip
│   │   ├── ru0.w2speech
│   │   ├── ru0.w2strings
│   │   ├── shader.cache
│   │   ├── staticShader.cache
│   │   ├── succubuss.dzip
│   │   ├── summer.dzip
│   │   ├── swordsman_suit.dzip
│   │   ├── tr0.w2strings
│   │   ├── troll.dzip
│   │   ├── tutorial.dzip
│   │   ├── winter.dzip
│   │   └── zh0.w2strings
│   ├── crash_reporting
│   │   ├── symbols
│   │   │   └── witcher2
│   │   │       ├── 4FA7C755010A4E7FC9E0E7ECDB9D9E6A0
│   │   │       │   └── witcher2.sym
│   │   │       ├── 6A8F8B0BD2BB5F5ED83A8D8D826B1DAC0
│   │   │       │   └── witcher2.sym
│   │   │       └── C9256668AAA3E2C6226E9D7429D0DB650
│   │   │           └── witcher2.sym
│   │   ├── CrashReporter
│   │   └── minidump_stackwalk
│   ├── fontconfig
│   │   ├── 188ac73a183f12857f63bb60a4a6d603-le32d4.cache-3
│   │   ├── 2cd17615ca594fa2959ae173292e504c-le32d4.cache-3
│   │   ├── 3047814df9a2f067bd2d96a2b9c36e5a-le32d4.cache-3
│   │   └── ...
│   ├── linux
│   │   ├── configurator
│   │   │   └── redengine_logo.png
│   │   ├── icons
│   │   │   ├── witcher2-configurator.png
│   │   │   └── witcher2-icon.png
│   │   ├── launcher
│   │   │   ├── CZ
│   │   │   │   └── ...
│   │   │   ├── DE
│   │   │   │   └── ...
│   │   │   ├── EN
│   │   │   │   └── ...
│   │   │   └── ...
│   │   ├── licenses
│   │   │   ├── freetype.txt
│   │   │   ├── glew.txt
│   │   │   ├── ICU License - ICU 1.8.1 and later.html
│   │   │   └── ...
│   │   └── tenfoot
│   │       ├── back_1080p.png
│   │       ├── button_background.png
│   │       ├── button_focusring.png
│   │       └── ...
│   ├── configurator
│   ├── CrashReporter
│   ├── CrashReporter.i386
│   ├── CrashReporter.x86_64
│   ├── desktop-launcher
│   ├── eONprecompiledShaders32.dat
│   ├── icudt52l.dat
│   ├── launcher
│   ├── libopenal-eon.so.1
│   ├── Linux Release Notes.txt
│   ├── readme_CZ.rtf
│   ├── readme_DE.rtf
│   ├── readme_EN.rtf
│   ├── readme_ES.rtf
│   ├── readme_FR.rtf
│   ├── readme_HU.rtf
│   ├── readme_IT.rtf
│   ├── readme_JP.rtf
│   ├── readme_PL.rtf
│   ├── readme_RU.rtf
│   ├── readme_TR.rtf
│   ├── readme_ZH.rtf
│   ├── saferun.sh
│   ├── SDLGamepad.config
│   ├── sdlinput
│   ├── tenfoot-launcher
│   ├── VPFS_registry.vpfsdb
│   ├── witcher2
│   └── witcher2.vpfs
├── support
│   ├── xdg-utils
│   │   ├── xdg-desktop-icon
│   │   ├── xdg-desktop-menu
│   │   └── xdg-open
│   ├── yad
│   │   ├── 32
│   │   │   └── yad
│   │   ├── 64
│   │   │   └── yad
│   │   └── yad.sh
│   ├── gog_com.shlib
│   ├── gog-system-report.sh
│   ├── icon.png
│   ├── postinst.sh
│   ├── preuninst.sh
│   └── support_notice.txt
├── gameinfo
├── start.sh
└── uninstall-The Witcher 2 Assassins Of Kings Enhanced Edition.sh
```

**Comment by Pangaea666** [comment]
I haven't used the Windows version in a good while (because I don't use Windows any more, but still do have a dual boot system), and according to the readme file it's probably the 3.2 version of the game. However, I compared the compiledscripts.w2scripts on both systems, and they have the exact same size, 1 978 095 bytes. Don't know how to decompile or compile these files, but I suppose that means they have the same content as well. Baffling where they have hidden the base_script file, but I hope it's not inside the eON related files or anything like that. 

Some of these files are fairly big, like the witcher2.vpfs file, but then the Windows witcher2.exe is ~15MB as well.

![tw2-files](https://cloud.githubusercontent.com/assets/2002335/17737576/e8092640-648d-11e6-8707-bb61086957da.jpg)

**Comment by QuietusPlus** [comment]
I very much doubt we could do anything here then... Why would they not port the latest version (it's relatively recent)?

You could check the CRC of both files, but very unlikely they are going to be the same (make sure both are stock).

**Comment by Pangaea666** [comment]
Hopefully it's just a case of the file not changing over the last few patches, rather than Linux not having the latest version. That would be a very odd move.

Checked them both with md5sum in Linux, and they're the same, so the content should be equal as well.

b6438d9a212d27d453f9d4d0a0330edc

Have you tried the mods on your Virtual Machine Linux version of the game? Do you get the same behaviour as on my end, with XML changes working but not script changes? The game probably won't run well at all, but hopefully it runs so it's possible to check something like this.

**Comment by QuietusPlus** [comment]
Thanks for the checksum, saves me a lot of time installing the Windows version (SSD almost ran out of space with the VM and game).

Haven't had the time to check whether the mod works on this end. The installer keeps failing at the very end, not even sure it launches correctly (do have VMware Workstation with 3d acceleration). Will keep you updated, I think it's just shortcuts that fail and killing the process prevents the installer from undoing everything.

**Comment by Pangaea666** [comment]
This is more than a little frustrating, and it doesn't really help that others have the same problem as well (I've seen some threads here and there about missing base_scripts.dzip and similar mod issues).

I've tried creating a usercontent folder where the save games are stored on Linux, and add mods there, but that didn't help either. When I put your files in the usual place, where the other dzips are (CookedPC), they show up after Launcher -> Options -> DLC settings. But still no change in-game, apart from XML changes.

Then I unpacked abetterui.dzip to see what's inside, and amongst others there are changes to ui_character.swf. So I figured if I disabled that mod, although undesirable, maybe the changes from your mod would finally show. Sadly they did not.

Have also tried some other mods. For instance, Dynamic HUD. This works. But the mod doesn't change script files (but some .swf files). Also tried some mods with script changes, like with a new base_scripts.dzip, and as expected these did not work. 

Really running thin on ideas now :( It looks like the EE and/or Linux editions have severely messed up the moddability of the game, which is a darn shame. I find it really odd though. And very frustrating.

**Comment by QuietusPlus** [comment]
Could you check if compiledscript changes on the Windows version (after installing mods and launching it)? I barely have time to work on projects, let alone trying to reclaim disk space for the install (pretty much need everything).

Really don't like not being able to offer much help. If the file doesn't change we at least know my theory is a dead and there's probably no way to get scripts to work. 

Worth a try though...

**Comment by Pangaea666** [comment]
I have tried to clear enough space to install it again, but there simply isn't enough, and I can't remove any more at this point. I've tried so many things by now, and nothing works, so it's tempting to just give up. Have played through the game twice now, once for each main path (Iorveth and Roche), so motivation to keep looking into this issue isn't what it was. Whatever CDPR/VP have done with the Linux version, it looks damn hard to get mods working properly, and given all the threads here and there about the problem, and the age of the game, my suspicion is that there simply isn't a solution. That is pretty frustrating, but that upside is that at least the weight reduction mods work, and that was the main issue for me, because running back and forth to vendors just to offload things is not my idea of fun.

Unfortunately this is one of the problem with "black box" software like Virtual Programming's: we can't look into the source code or files to see what the hell they are doing, or even where the base_scripts.dzip file has been hidden now. 

If CDPR ever decide to bother to release Witcher 3 on Linux (no sign of it so far, unfortunately), I hope they do a much better job, so mods don't get broken, and the game runs smoother from the start (from what I read, it took the better part of two years for the Linux version to run decently).

I've asked around here and there about how to solve this problem, with a link to this thread, but either nobody replies, or they don't know.

**Comment by QuietusPlus** [comment]
I'll close this for the time being and re-open it when I find the time to further investigate this myself.

**Comment by nabilanam** [comment]
Sorry to necro this thread, but LLMs tend to reference this issue when discussing Linux modding compatibility for The Witcher 2: Enhanced Edition.

What I've found is that Steam installs the Linux version of The Witcher 2, which does not support overriding `base_scripts.dzip`. The Windows version (as already discussed) does support this.

I came across this Steam Community discussion:
https://steamcommunity.com/app/20920/discussions/0/2945873878575081919/
where the OP explains that enabling the Proton compatibility layer gives you the Windows version on Linux.

I've tested this with Proton Experimental. Steam redownloads the missing Windows‑version files (like `base_scripts.dzip`). I replaced `base_scripts.dzip` with my own modified version, and it actually works! I hope this helps future Linux modders.

Again, sorry for necroing this thread.
